### ISSUE TITLE
val loss collapses but not training loss

### ISSUE THREAD

**Issue by AceMcAwesome** [qubvel/segmentation_models#360]
Hi, I am training a 4-output segmentation model using UNet and inceptionresnetv2.  My loss has been behaving strangely in that as my training loss for one or more classes starts to improve at a greater rate, the val loss will suddenly go to near zero despite the training loss being fine.  I also manually calculate the dice score for each class individually and print it below the epoch training, and whenever the val loss goes to near zero it corresponds with one of my class's dice scores going to exactly zero.  See below an example of this happening:

STARTING EPOCH 17  lr 0.0025 epoch_time 506.0
500/500 [==============================] - 455s 909ms/step - loss: 0.8594 - f1-score: 0.2128 - val_loss: 0.8598 - val_f1-score: 0.2123
dice_scores 0.0012828 masks_1
dice_scores 0.000868 masks_2
dice_scores 0.0003565 masks_3
dice_scores 0.8639288 masks_4
STARTING EPOCH 18  lr 0.0025 epoch_time 519.0
500/500 [==============================] - 451s 903ms/step - loss: 0.8569 - f1-score: 0.2166 - val_loss: 0.8547 - val_f1-score: 0.2201
dice_scores 0.0012946 masks_1
dice_scores 0.0008772 masks_2
dice_scores 0.0003479 masks_3
dice_scores 0.8896476 masks_4
STARTING EPOCH 19  lr 0.0025 epoch_time 506.0
500/500 [==============================] - 455s 910ms/step - loss: 0.8609 - f1-score: 0.2106 - val_loss: 0.9994 - val_f1-score: 5.3482e-04
dice_scores 0.0012285 masks_1
dice_scores 0.0008384 masks_2
dice_scores 0.000348 masks_3
dice_scores 0.0 masks_4

I understand that gradients can collapse when the learning rate is too high.  However, I mainly don't understand why the training accuracy would not also collapse then?  If the validation part is just running the same weights as the training accuracy, why does training accuracy for that epoch remain fine while validation accuracy goes to zero?

My solution to this has been to check if any validation dice scores go to exactly zero after an epoch, and if so reload the model from the last saved good point and reduce the learning rate by half, and try again.  This works but results in slower model training, and I would like to understand what is actually happening when this validation-only collapse happens.  Thanks!
